pkg/otlp/collector.go

```diff
@@ -206,6 +206,14 @@ func NewPipeline(cfg PipelineConfig, s serializer.MetricSerializer, logsAgentCha
 
 // Run the OTLP pipeline.
 func (p *Pipeline) Run(ctx context.Context) error {
+	defer func() {
+		// Recover any panic that happened during execution.
+		if r := recover(); r != nil {
```

Comment:
Is there a unit test we can write to verify this recover works as intended?

Reply:
I moved this to a separate function and added a test for that, since I can't test directly the `Run` method with this (no good way to trigger a panic)